lib/models/NotificationConfiguration.js

```diff
@@ -0,0 +1,275 @@
+  *         queueArn: string,
+  *         filterRules: [
+  *             {
+  *                 name: string,
+  *                 value: string
+  *             },
+  *             {
+  *                 name: string,
+  *                 value:string
+  *             },
+  *         ],
+  *         id: string
+  *         }
+  *     ]
+  * }
+  */
+
+class NotificationConfiguration {
+    /**
+     * Create a Notification Configuration instance
+     * @param {string} xml - parsed configuration xml
+     * @return {object} - NotificationConfiguration instance
+     */
+    constructor(xml) {
+        this._parsedXml = xml;
+        this._config = {};
+        this._ids = [];
+    }
+
+    /**
+     * Get notification configuration
+     * @return {object} - contains error if parsing failed
+     */
+    getValidatedNotificationConfiguration() {
+        const validationError = this._parseNotificationConfig();
+        if (validationError) {
+            this._config.error = validationError;
+        }
+        return this._config;
+    }
+
+    /**
+     * Check that notification configuration is valid
+     * @return {error | undefined} - error if parsing failed, else undefined
+     */
+    _parseNotificationConfig() {
+        if (!this._parsedXml || this._parsedXml === '') {
+            return errors.MalformedXML.customizeDescription(
+                'request xml is undefined or empty');
+        }
+        const notificationConfig = this._parsedXml.NotificationConfiguration;
+        if (!notificationConfig || notificationConfig === '') {
+            return errors.MalformedXML.customizeDescription(
+                'request xml does not include NotificationConfiguration');
+        }
+        const queueConfig = notificationConfig.QueueConfiguration;
+        if (!queueConfig || !queueConfig[0] || queueConfig[0] === '') {
+            return errors.MalformedXML.customizeDescription(
+                'request xml does not include QueueConfiguration');
+        }
+        this._config.queueConfig = [];
+        let parseError;
+        queueConfig.forEach(c => {
+            const eventObj = this._parseEvents(c.Event);
+            const filterObj = this._parseFilter(c.Filter);
+            const idObj = this._parseId(c.Id);
+            const arnObj = this._parseArn(c.QueueArn);
+
+            if (eventObj.error) {
+                parseError = eventObj.error;
+            } else if (filterObj.error) {
+                parseError = filterObj.error;
+            } else if (idObj.error) {
+                parseError = idObj.error;
+            } else if (arnObj.error) {
+                parseError = arnObj.error;
+            } else {
+                this._config.queueConfig.push({
+                    events: eventObj.events,
+                    queueArn: arnObj.arn,
+                    id: idObj.id,
+                    filterRules: filterObj.filterRules,
+                });
+            }
+        });
+        return parseError;
```

Comment:
The logic here will override errors as you move through, additionally even if something does have an error `_config` will be changed in the process. Not a fan of the methods having (in my opinion hidden) side effects either.

Reply:
If there are any errors, `_config` won't be changed. And yeah, the errors will overwrite each other if there are multiple, but since we're trying to return more specific error messages, it doesn't really make sense to combine them, and it doesn't really matter which one gets returned first

Reply:
> If there are any errors, `_config` won't be changed. 

Per iteration in the loop yes, but ( and I am not sure this matters ) if the first was valid and the second item in the list was an error then `_config` would be changed and an error returned.

Reply:
oh, you're totally right, didn't even think of that. i'll reset `this._config` if there's an error

Reply:
I believe `_id` is also changed as part of the `_parseId` method but don't take my word for it.

Reply:
I'll take a look. Any changes I make to this class are going to be in this PR tho: https://github.com/scality/Arsenal/pull/1231
The diff is in this PR because currently this branch is based off of that one until it gets merged.

Reply:
ok no worries, feel free to point me somewhere else if I do that again.